src/plugins/GUITestBase/src/GTUtilsPcrPrimerDesignForDnaAssembly.cpp

```diff
@@ -0,0 +1,367 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include "GTUtilsPcrPrimerDesignForDnaAssembly.h"
+
+#include <QGroupBox>
+#include <QTableWidget>
+
+#include "GTUtilsOptionPanelSequenceView.h"
+#include "GTUtilsSequenceView.h"
+#include "GTUtilsTaskTreeView.h"
+
+#include "base_dialogs/GTFileDialog.h"
+
+#include "drivers/GTMouseDriver.h"
+
+#include "primitives/GTLineEdit.h"
+#include "primitives/GTRadioButton.h"
+#include "primitives/GTSpinBox.h"
+#include "primitives/GTTableView.h"
+#include "primitives/GTWidget.h"
+
+namespace U2 {
+using namespace HI;
+
+#define GT_CLASS_NAME "GTUtilsPcrPrimerDesign"
+
+static const QString badBackboneDialogName = "UnwantedStructuresInBackboneDialog";
+
+// Parent of most widgets used in this class. Doesn't open the tab. Returns nullptr in case of an error.
+static QWidget* getTabWgt(GUITestOpStatus& os) {
+    const QWidget* inner = GTWidget::findWidget(os,
+        GTUtilsOptionPanelSequenceView::innerWidgetNames[GTUtilsOptionPanelSequenceView::PcrPrimerDesign],
+        GTUtilsSequenceView::getActiveSequenceViewWindow(os));
+    return GTWidget::findWidget(os, "runPcrPrimerDesignWidget", inner);
+}
+
+// Return the sequence generator widget. Doesn't open the tab. Returns nullptr in case of an error.
+static QWidget* getGenSeqWgt(GUITestOpStatus& os, const QWidget* tab) {
+    return GTWidget::findWidget(os, "wgtGenerateSequence", tab == nullptr ? getTabWgt(os) : tab);
+}
+
+// Return the generated sequence table. Doesn't open the tab. Returns nullptr in case of an error.
+static QTableWidget* getGenSeqTable(GUITestOpStatus& os, const QWidget* tab) {
+    return GTWidget::findExactWidget<QTableWidget*>(os, "twGeneratedSequences", getGenSeqWgt(os, tab));
+}
+
+// Return the table of result sequences. Doesn't open the tab. Returns nullptr in case of an error.
+static QTableWidget* getResultTable(GUITestOpStatus& os, const QWidget* tab) {
+    return GTWidget::findExactWidget<QTableWidget*>(os, "productsTable", tab);
+}
+
+/**
+ * Moves the mouse pointer over a label, then scrolls in the required direction so that the specified
+ * widget ('scrollTo') is displayed in full. If the widget isn't displayed after scrolling, sets an operation error.
+ * scrollTo should not be nullptr.
+ *
+ * At the moment the scroll doesn't work if the tasks/log are open.
+ */
+#define GT_METHOD_NAME "scrollToWidget"
+static void scrollToWidget(GUITestOpStatus& os, const QWidget* scrollTo, const QWidget* tab) {
+    GT_CHECK(scrollTo != nullptr, "Can't scroll to nullptr")
+    const auto isWidgetFullyVisible = [](const QWidget* w) {
+        const QRegion visibleRegion = w->visibleRegion();
+        return !visibleRegion.isEmpty() && visibleRegion.boundingRect().height() == w->height();
+    };
+    const auto moveMouseAndScroll = [](QWidget* moveMouseTo, bool scrollUp) {
+        GTMouseDriver::moveTo(GTWidget::getWidgetCenter(moveMouseTo));
+        const int scrollCoef = 20;
```

Comment:
Dont use consts for locals (you do not use them consistently in this CL and UGENE does not use them across the code too)